### ISSUE TITLE
destructuring optional with default before non-optional parameter should not be allowed.

### ISSUE THREAD

**Issue by shaunc** [Microsoft/TypeScript#20209]
**TypeScript Version:**  2.6.1
 
If an optional destructured argument with a default is followed by another parameter with 
no default, an error should be raised, as, to make the call, the first one must be specified
in order to specify the parameter following. Instead, no error is generated, but the resulting
declaration file is not usable.

**Code**

```ts
function foo ({a}: {a?: string} = {}) {}
function foo2 ({a}: {a?: string} = {}, b: string) {}
```

**Expected behavior:**

Error.

**Actual behavior:**

Declaration file:
```ts
declare function foo({a}?: {
    a?: string;
}): void;
declare function foo2({a}: {
    a?: string | undefined;
} | undefined, b: string): void;
```
When consuming this declaration file, `foo2` is flagged as an error.
